File: cses/01_introductory/3419_mex-grid-construction/solution.py
import sys


def main():
    data = sys.stdin.buffer.read().split()
    n = int(data[0])

    # Giá trị mex tại ô (i, j) theo 0-indexed chính là i XOR j.
    # Tạo chuỗi từng hàng bằng join rồi gom mọi hàng vào danh sách out,
    # cuối cùng ghi ra một lần (không dựng mảng lưới 2 chiều).
    out = []
    for i in range(n):
        out.append(" ".join(str(i ^ j) for j in range(n)))

    sys.stdout.write("\n".join(out))
    # Thêm ký tự xuống dòng kết thúc, bỏ qua khi n = 0 (không có hàng nào).
    if n > 0:
        sys.stdout.write("\n")


main()

File: cses/01_introductory/3419_mex-grid-construction/solution.cpp
#include <iostream>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int n;
    cin >> n;

    // Giá trị mex tại ô (row, col) theo 0-indexed chính là row XOR col.
    // In từng hàng: các giá trị cách nhau bởi dấu cách, mỗi hàng một dòng.
    for (int row = 0; row < n; ++row) {
        for (int col = 0; col < n; ++col) {
            if (col > 0) {
                cout << ' ';
            }
            cout << (row ^ col);
        }
        cout << '\n';
    }

    return 0;
}
